src/components/TodoItem.vue

```diff
@@ -0,0 +1,54 @@
+<template>
+  <li :style="[item.done ? {'background-color':'#698aab'} : {'background-color':'#a0d8ef'}]">
+    <label v-if="!isEditing">{{ item.title }}</label>
+    <b-form-input
+      v-else
+      v-model="newTitle"
+      :placeholder="[[item.title]]"
+    />
+
+    <b-button @click="editTitle(newTitle)">
+      <b-icon icon="pencil-fill" />
+    </b-button>
+
+    <label>{{ item.date }}</label>
+
+    <span>
+      <b-button @click="toggleDone(index)">
+        {{ item.done ? "Incompleted" : "Completed" }}
+      </b-button>
+      <b-button
+        variant="danger"
+        @click="$emit('remove', index)"
```

Comment:
We can wrap the code with a method without a JavaScript expression in a line.
On the other hand, we can just commit to mutation directly since we have implemented method in vuex that we don't have to pass event to parent component.